x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/feature_tour/rules_feature_tour.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiStatelessTourStep,
+  EuiText,
+  EuiTourActions,
+  EuiTourState,
+  EuiTourStep,
+  EuiTourStepProps,
+  useEuiTour,
+} from '@elastic/eui';
+import { noop } from 'lodash';
+import React, { FC, useEffect, useMemo, useState } from 'react';
+import { RULES_MANAGEMENT_FEATURE_TOUR_STORAGE_KEY } from '../../../../../../../common/constants';
+import { useKibana } from '../../../../../../common/lib/kibana';
+import * as i18n from './translations';
+
+export interface RulesFeatureTourContextType {
+  steps: EuiTourStepProps[];
+  actions: EuiTourActions;
+}
+
+export const SEARCH_CAPABILITIES_TOUR_ANCHOR = 'search-capabilities-tour-anchor';
+
+const TOUR_POPOVER_WIDTH = 400;
+
+const tourConfig: EuiTourState = {
+  currentTourStep: 1,
+  isTourActive: true,
+  tourPopoverWidth: TOUR_POPOVER_WIDTH,
+  tourSubtitle: i18n.TOUR_TITLE,
+};
+
+const stepsConfig: EuiStatelessTourStep[] = [
+  {
+    step: 1,
+    title: i18n.SEARCH_CAPABILITIES_TITLE,
+    content: <EuiText>{i18n.SEARCH_CAPABILITIES_DESCRIPTION}</EuiText>,
+    stepsTotal: 1,
+    children: <></>,
+    onFinish: noop,
+    maxWidth: TOUR_POPOVER_WIDTH,
+  },
+];
+
+export const RulesFeatureTour: FC = () => {
+  const { storage } = useKibana().services;
+
+  const restoredState = useMemo<EuiTourState>(
+    () => ({
+      ...tourConfig,
+      ...storage.get(RULES_MANAGEMENT_FEATURE_TOUR_STORAGE_KEY),
+    }),
+    [storage]
+  );
+
+  const [tourSteps, tourActions, tourState] = useEuiTour(stepsConfig, restoredState);
+
+  useEffect(() => {
+    const { isTourActive, currentTourStep } = tourState;
+    storage.set(RULES_MANAGEMENT_FEATURE_TOUR_STORAGE_KEY, { isTourActive, currentTourStep });
+  }, [tourState, storage]);
+
+  const [shouldShowSearchCapabilitiesTour, setShouldShowSearchCapabilitiesTour] = useState(false);
+
+  useEffect(() => {
+    /**
+     * Wait until the tour target elements are visible on the page and mount
+     * EuiTourStep components only after that. Otherwise, the tours would never
+     * show up on the page.
+     */
+    const observer = new MutationObserver(() => {
+      if (document.querySelector(`#${SEARCH_CAPABILITIES_TOUR_ANCHOR}`)) {
+        setShouldShowSearchCapabilitiesTour(true);
+        observer.disconnect();
+      }
```

Comment:
How many times it can be called before the target element shows up in the `document`?
Should we debounce this function?

Reply:
It depends on the number of re-renders. I see this function being called ~5 times on standard navigation and ~10-15 times on full page reload. But I want to mention that DOM element selectors by id are super fast. I ran a couple of tests, and the performance tab showed me ~30μs per call. So it's not worth optimization, IMO.